pyproject.toml

```diff
@@ -2,7 +2,7 @@
 build-backend = "mesonpy"
 requires = [
     "meson-python>=0.15.0",
-    "Cython>=3.0",
+    "Cython>=3.0.6",  # keep in sync with version check in meson.build
```

Comment:
Should `build_requirements.txt` also be updated with this version?

Reply:
Good catch, updated.

Reply:
And maybe also update `environment.yml`?

Reply:
No need I think, we don't add lower bounds or pins there unless there's an actual problem to resolve. Which won't be the case here, it should always pick the most recent Cython version.

Reply:
Oh okay, I though since there's a pin for `>=3.0`, an update might be needed.

Reply:
I read that more as "no longer supports 0.29.X" - we should be able to drop that `<3.0` at some point, once other relevant packages that one may want to install into the same env have transitioned.